### PR TITLE
fix(43535): TypeScript preserves 'override' modifier in JavaScript output

### LINKED ISSUE #43535: TypeScript preserves 'override' modifier in JavaScript output
I'm not sure where to ask this, but it seems like `tsc` is compiling and leaving the `override` keyword in the compiled result. Is this something other folks are seeing? It seems like it needs the signature to match exactly, or it will show up in the output, see here:

https://www.typescriptlang.org/play?ts=4.3.0-beta#code/LAKAxgNghgzjAEBlADgUwheBvUp72QFcAjCASzHgHNUAXAYQAsoAnGmACgEoAueAO0IBbYqhbY8+eCzqEW-eAAYA3JIC+uEPiKkK1OgBE5UWmQD2-bn0EixErVJm05Cles3aS5SmFi0r8ABuZmQAJtgaIJGgkLAIABJmEACeAILG8KgAHrSo-KEIKOiYOCCSZoFiLGGo+gzMbKicXPZS0rLy8ACMbg7RDhVVNXVGLCbmlrwCwqLipW1OLvAAzL34-fiDLNWhtb4w-lPBYa3w-WpAA

_Originally posted by @seiyria in https://github.com/microsoft/TypeScript/issues/2000#issuecomment-813445255_

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #43535

### CODE REVIEW

**Review by DanielRosenwasser — APPROVED** [review]

**Review by Kingwl — APPROVED** [review]

**Comment by Kingwl** [comment]
Thanks!

File: tests/baselines/reference/override12(target=esnext).js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    public m2(): number {
+        return 0;
+    }
+
+    public m3(): void {}
+}
+
+class B extends A {
+    override m1() {
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/override12(target=esnext).js:15]: 🤦‍♂️

Comment (a-tarasyuk) [reply tests/baselines/reference/override12(target=esnext).js:15]: @DanielRosenwasser ?

Comment (Kingwl) [reply tests/baselines/reference/override12(target=esnext).js:15]: Holy shit🤦‍♂️

Comment (a-tarasyuk) [reply tests/baselines/reference/override12(target=esnext).js:15]: @Kingwl @DanielRosenwasser Could you clarify this case? `//// [override12.js]` does not contain `override` keywords.

Comment (Kingwl) [reply tests/baselines/reference/override12(target=esnext).js:15]: But if `useDefineForClassFields` is on....

[playground](https://www.typescriptlang.org/play?useDefineForClassFields=true&target=99&noImplicitOverride=true&ts=4.4.0-dev.20210526#code/MYGwhgzhAEDC0G8BQ1XTNAvNAjEgvkkqJDACLQCmAHgC6UB2AJjPMmtAPYBulATnwCWTSuizQATASA)

**Pull request merged**